types/src/v0/v0_3/stake_table.rs

```diff
@@ -56,3 +59,33 @@ pub type IndexedStake = (
     EpochNumber,
     IndexMap<alloy::primitives::Address, Validator<BLSPubKey>>,
 );
+
+
+
+#[derive(Clone, derive_more::derive::Debug)]
+pub struct StakeTableFetcher {
+    /// Peers for catching up the stake table
+    #[debug(skip)]
+    pub(crate) peers: Arc<dyn StateCatchup>, 
+    /// Methods for stake table persistence.
+    #[debug(skip)]
+    pub(crate)  persistence: Arc<dyn MembershipPersistence>,
+    /// L1 provider
+    pub(crate)  l1_client: L1Client,
+    /// Verifiable `ChainConfig` holding contract address
+    pub(crate) chain_config: ChainConfig,
+    /// Flag whether to store the fetched events from contract
+    pub (crate) store_events: bool,
```

Comment:
Do we need this flag? Seems a bit strange that we have a persistence member and *also* have this flag.